rootfs/scheduler/__init__.py

```diff
@@ -1094,6 +1098,42 @@ def _buildpack_readiness_probe(self, controller, delay=30, timeout=5, period_sec
 
         return controller
 
+    '''
+    Applies tcp socket readiness probe to the docker app container only if some port is exposed
+    by the docker image.
+    '''
+    def _dockerapp_readiness_probe(self, controller, image, delay=5, timeout=5, period_seconds=5,
+                                   success_threshold=1, failure_threshold=1):
+        port = self._get_port(image)
+        if port is None:
+            return controller
+        readinessprobe = {
```

Comment:
Can we move a lot of this logic into a more generic function like `_readiness_probe`? There's a significant amount of duplicate code here with `_buildpack_readiness_probe()`. Normally I'd say it's fine but there is a significant amount of shared code here.